### ISSUE TITLE
Spurious "Invalid connection credentials sent by remote" errors

### ISSUE THREAD

**Issue by fredcallaway** [JuliaLang/julia#39871]
If I leave a julia process running overnight on a 96-core machine I get about 100 copies of this error message:

```
      From worker 44:   ErrorException("Process(44) - Invalid connection credentials sent by remote.")CapturedException(ErrorException("Process(44) - Invalid connection credentials sent by remote."), Any[(error(::String) at error.jl:33, 1), (process_hdr(::Sockets.TCPSocket, ::Bool) at process_messages.jl:273, 1), (message_handler_loop(::Sockets.TCPSocket, ::Sockets.TCPSocket, ::Bool) at process_messages.jl:167, 1), (process_tcp_streams(::Sockets.TCPSocket, ::Sockets.TCPSocket, ::Bool) at process_messages.jl:142, 1), ((::Distributed.var"#99#100"{Sockets.TCPSocket,Sockets.TCPSocket,Bool})() at task.jl:356, 1)])                                       
      From worker 44:   Process(44) - Unknown remote, closing connection.
```

However, I still have 96 processes running. I think this message should probably not be printed if the process is able to reconnect. As a short-term fix, I would like to silence this error message but, I can't find any way to do so. Supressor.jl is not good because I only want to suppress this one message.

I'm using Julia 1.5.0 on a machine running a Red Hat spinoff (Springdale Linux 7.6).

**Comment by c42f** [comment]
It looks to me that there's no good way to silence this.

Perhaps you could add some logging to the Distributed standard library so we can dig to the root cause a bit more? It seems this error originates from the `process_hdr()` function in Distributed.jl. So for instance, logging the `self_cookie` and `cookie` might provide some insight.

The logs suggest we end up here

https://github.com/JuliaLang/julia/blob/696cb1ba7bf03b7c275a870e5f4229c986e1383f/stdlib/Distributed/src/process_messages.jl#L212

At that point we can't even call `deregister_worker()` because the connection was never initialized properly. So it could be that you've got several Julia worker processes hanging around which never joined the cluster successfully.

Which of cluster manager are you using and how are you starting it? I assume you're using the builtin `LocalManager`?

**Comment by fredcallaway** [comment]
Sorry for the delay. I've been trying to compile from source to add logging to base but haven't been able to (not sure I should raise an issue because it's probably specific to me). Anyway, I think it will probably work fine to just redefine this function in my user code. I'll give that a shot.

Yes, I'm using `LocalManager` via `addprocs()`. 

>  So it could be that you've got several Julia worker processes hanging around which never joined the cluster successfully.

That's not ideal. To test for this, I can kill all my julia processes, then start exactly 40 of them and see if there are more than 40 (using `htop`) after I get the invalid credentials message. Does that sound right?

**Comment by c42f** [comment]
> To test for this, I can kill all my julia processes, then start exactly 40 of them and see if there are more than 40 (using htop) after I get the invalid credentials message. Does that sound right?

Yes, you should start with zero julia processes for clarity.  Though you could see a tree of all processes with julia highlighted (and hence the parent-child relationship with the child processes) using something like

```bash
pstree -g -a | egrep 'julia|$'
```

But I think you want to add processes manually and somehow test whether the expected number are visible from the master node. Instead of the `-p` argument to Julia, you can use something like:

```
julia> using Distributed

julia> manager = Distributed.LocalManager(4, true)
LocalManager()

julia> workers()
1-element Vector{Int64}:
 1

julia> addprocs(manager)
4-element Vector{Int64}:
 2
 3
 4
 5

julia> workers()
4-element Vector{Int64}:
 2
 3
 4
 5
```

Do the expected number of workers show up?  Can you connect to each of them and retrieve data? For example,

```julia
julia> @fetchfrom 2 myid()
2
```

**Comment by fredcallaway** [comment]
OK I ran these tests. The expected number of workers show up and all of them can send/receive data.

Additionally, after 48 hours or so, all of the workers have produced the invalid connection warning exactly twice each. However, they still work fine and the process ids have not changed.

Interestingly, the messages come in numerical order both times, perhaps suggesting that an OS process is periodically closing connections when they reach a certain age.

I'll try adding the logging next.
